core/store/src/archive/cloud_storage/get.rs

```diff
@@ -44,6 +45,23 @@ impl CloudStorage {
             block_on_future(self.retrieve_block_data(block_height)).map_err(Error::other)?;
         Ok(block_data)
     }
+
+    #[allow(unused)]
+    fn get_shard_data(
+        &self,
+        hot_db: &Arc<dyn Database>,
+        block_hash: &[u8],
+        shard_id: near_primitives::types::ShardId,
+    ) -> Result<ShardData> {
+        let block_hash = CryptoHash::try_from(block_hash)
+            .map_err(|error| Error::new(ErrorKind::InvalidInput, error))?;
+        let hot_store = Store::new(hot_db.clone());
```

Comment:
Should we hold an instance of `hot_store` in the struct instead of creating it on the fly every time?